source/unix/openssl_platform_init.c

```diff
@@ -92,24 +94,55 @@ static int s_hmac_ctx_reset(HMAC_CTX *ctx) {
     return 1;
 }
 
+static struct aws_mutex *s_libcrypto_locks = NULL;
+static struct aws_allocator *s_libcrypto_allocator = NULL;
+
+static void s_locking_fn(int mode, int n, const char *unused0, int unused1) {
+    (void)unused0;
+    (void)unused1;
+
+    if (mode & CRYPTO_LOCK) {
+        aws_mutex_lock(&s_libcrypto_locks[n]);
+    } else {
+        aws_mutex_unlock(&s_libcrypto_locks[n]);
+    }
+}
+
+static unsigned long s_id_fn(void) {
+    return (unsigned long)aws_thread_current_thread_id();
+}
+
+enum aws_libcrypto_version {
+    AWS_LIBCRYPTO_NONE,
+    AWS_LIBCRYPTO_102,
+    AWS_LIBCRYPTO_111,
```

Comment:
this is confusing me.
how is "102" == "1.0.0"?
in any case it would probably help my understanding if we separated major/minor/patch like `AWS_LIBCRYPTO_1_1_1`

Reply:
1.0.0 in the so-name is the ABI revision, not the software version. I'll fix the enum identifiers though.